src/Radicle/Internal/Effects/Capabilities.hs

```diff
@@ -137,8 +140,46 @@ instance ReadFile IO where
 instance {-# OVERLAPPABLE #-} ReadFile m => ReadFile (Lang m) where
     readFileS = lift . readFileS
 
+class (Monad m) => FileModule m where
+    fileModuleS :: Text -> m (Either Text Text)
```

Comment:
I would prefer `m (Maybe Text)` as the return type and let the caller handle missing files.

Reply:
Done.